compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -2753,6 +2753,19 @@ class Typer extends Namer
         tree.tpe <:< wildApprox(pt)
         readaptSimplified(Inliner.inlineCall(tree, pt))
       }
+      else if (tree.symbol.isScala2Macro) {
+        if (ctx.settings.XignoreScala2Macros.value) {
+          ctx.warning("Scala 2 macro cannot be used in Dotty. See http://dotty.epfl.ch/docs/reference/dropped-features/macros.html", tree.sourcePos)
+          tree
+        } else if (tree.symbol eq defn.StringContext_f) {
```

Comment:
Add some documentation explaining what this branch does exactly and what the assumptions are.

Reply:
Modified the documentation.